plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/WarPublisher.java

```diff
@@ -80,7 +81,10 @@ public static void publishWar(IProject project, IPath destination, IPath safeWor
 
     IModuleResource[] resources =
         flattenResources(project, safeWorkDirectory, subMonitor.newChild(10));
-    PublishUtil.publishZip(resources, destination, subMonitor.newChild(90));
+    if (resources.length == 0) {
+      logger.log(Level.WARNING, "no resource to publish");
```

Comment:
If this is an issue should we return an error status immediately and not call publishZip?

Reply:
Makes sense. There must be something to publish all the time.